packages/fxa-payments-server/src/routes/Subscriptions/PaymentUpdateForm.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useCallback, useEffect } from 'react';
+import { useBooleanState } from '../../lib/hooks';
+import { injectStripe, CardElement, ReactStripeElements } from 'react-stripe-elements';
+import { UpdatePaymentFetchState, CustomerFetchState } from '../../store/types';
+
+type PaymentUpdateFormProps = {
+  accessToken: string,
+  customer: CustomerFetchState,
+  resetUpdatePayment: Function,
+  updatePayment: Function,
+  updatePaymentStatus: UpdatePaymentFetchState
+};
+export const PaymentUpdateForm = ({
+  accessToken,
+  updatePayment,
+  updatePaymentStatus,
+  resetUpdatePayment,
+  customer,
+  stripe
+}: PaymentUpdateFormProps & ReactStripeElements.InjectedStripeProps) => {
+  const [ updateRevealed, revealUpdate, hideUpdate ] = useBooleanState();
+
+  // Reset payment update status on initial render.
+  useEffect(() => { 
+    resetUpdatePayment(); 
+  }, [ resetUpdatePayment ]);
+
+  const onSubmit = useCallback(ev => {
+    ev.preventDefault();
+
+    // TODO: use react state on form fields along with validation
+    const data = new FormData(ev.target);
+    const name = String(data.get('name'));
+
+    if (stripe) {
+      stripe
+        .createToken({ name })
+        .then((result) => {
+          updatePayment(accessToken, {
+            paymentToken: result && result.token && result.token.id,
+          });
+          hideUpdate(ev);
+        });
+        // TODO: error handling
+    }
+  }, [ accessToken, updatePayment, stripe ]);
+
+  if (customer.loading) {
+    // If the customer details are loading, then we have nothing to update yet.
+    return <span></span>;
```

Comment:
Why `<span />` instead of null?

Reply:
Oh hey, null works here! I'll do that instead. Originally, I tried an empty string, but TypeScript raised an error, so `<span />` was a shrug I forgot about